velox/common/memory/MappedMemory.h

```diff
@@ -190,13 +241,36 @@ class MappedMemory {
   // Returns the number of freed bytes.
   virtual int64_t free(Allocation& allocation) = 0;
 
+  // Makes a contiguous mmap of 'numPages'. Advises away the required
+  // number of free pages so as not to have resident size exceed the
+  // capacity if capacity is bounded. Returns false if sufficient free
+  // pages do not exist. 'collateral' and 'allocation' are freed and
+  // unmapped or advised away to provide pages to back the new
+  // 'allocation'. This will always succeed if collateral and
```

Comment:
> This will always succeed

It is a bit unclear why this is the case? Don't we need a contiguous chunk of memory, not just enough total memory across the specified allocations?

Reply:
In comment